handout/architect/10_ways_to_misuse_angular.md

```diff
@@ -0,0 +1,439 @@
+
+The card component below renders a heading and a paragraph. In order to control their layout we need a wrapper element. You might be tempted to wrap them in a `<div>`, but that is not required. Angular will render this with a `<my-card>` wrapper element. This is called a host element.
+
+```ts
+import { Component } from '@angular/core';
+
+@Component({
+  selector: 'my-card',
+  template: `
+  <h1 class="card-title">{{ title }}</h1>
+  <p class="card-body">
+    <ng-content></ng-content>
+  </p>
+  `
+})
+export class CardComponent { title = 'Title of card'; }
+```
+
+We can style the host element by targeting it with the `:host` selector in the component's CSS file. Alternatively you can attach CSS classes to it using [host metadata](https://angular.io/docs/ts/latest/cookbook/ts-to-js.html#!#host-metadata).
+
+```ts
+@Component({
+  selector: 'my-card',
+  host: { class: 'db ma0 lh-title' },
+  ...
+})
+export class CardComponent { title = 'Title of card'; }
+```
+
+We can use host metadata to bind to all properties and events of the host element. Alternatively, you can use the `@HostBinding` and `@HostListener` decorators.
+
+### Metadata vs decorators
+Host metadata and `@Host*` decorators can generally be used interchangeably however, there are some subtle differences. Consider the following scenario:
+
+```ts
+@Component({
+  selector: 'my-foo',
+  host: { 'class': 'navy' },
+})
+export class FooComponent {}
+
+@Component({
+  selector: 'my-bar',
+})
+export class BarComponent {
+  @HostBinding('class') className = 'navy';
+}
+```
+
+and somewhere later in our app you use them as:
+
+```html
+<my-foo class="underline"></my-foo>
+<my-bar class="underline"></my-bar>
+```
+
+`<my-foo>` will render with `class="underline navy"` however, `<my-bar>` will render with `class="navy"`. This is because `@HostBinding` only supports binding of values. Whereas, host metadata supports both binding and static values. In this case we are using a static value. Therefore, it merges what we have defined in the metadata with what is defined on the component. If we had bound class to a component property then they would behave the same:
+
+```ts
+@Component({
+  selector: 'my-foo',
+  host: { '[class]': 'className' },
+})
+export class FooComponent {
+  className = 'navy';
+}
+```
+
+
+
+## 2. Avoiding observables
+When working with Angular you should be using observables as much as possible. Angular exposes HTTP requests, form events, route params, etc. as observables. Therefore, by using observables we can build apps that have a consistent architectural pattern.
+
+### Leverage Observable Operators
+We can leverage observable operators to compute data based on application state. This allows us to reduce the amount of state we need to maintain. Additionally these computations can expressed as pure functions which makes testing easier. These pure play a similar to role to [selectors](https://github.com/reactjs/reselect) and can be memoized.
```

Comment:
s/these computations can expressed/these computations can be expressed/
s/These pure play/These pure functions play/